newIDE/app/src/PropertiesEditor/PropertiesMapToSchema.js

```diff
@@ -220,7 +220,40 @@ const createField = (
       getLabel,
       getDescription,
     };
-  } else {
+  } else if(valueType ==='animationname')
+    {
+      function getChoices()
+    {
+      if(!object)
+      { return [{value:"Object is not valid !", label:"Object is not valid !"}] }
+
+      let animationArray = [];
+      for(let i = 0; i < object.getConfiguration().getAnimationsCount(); i++ )
+      {
+          animationArray.push(object.getConfiguration().getAnimationName(i));
+      }
+        return animationArray.map(value => ({ value, label: value }));
+   
+    }
+      return {
+        name,
+        valueType: 'string',
+        getValue: (instance: Instance): string => {
+          return getProperties(instance)
+            .get(name)
+            .getValue();
+        },
+        setValue: (instance: Instance, newValue: string) => {
+          console.log(instance, name, newValue);
+          onUpdateProperty(instance, name, newValue);
+        },
+        getLabel,
+        getDescription,
+        getChoices,
```

Comment:
I guess a new attribute `isAllowingFreeText` could be added and `PropertiesEditor` will only allow the current property value until we make a new component that does autocompletion one day.